File: abstractions/java/lib/src/main/java/com/microsoft/kiota/authentication/AnonymousAuthenticationProvider.java
package com.microsoft.kiota.authentication;

import com.microsoft.kiota.RequestInformation;

import java.util.Map;
import java.util.concurrent.CompletableFuture;

/** This authentication provider does not perform any authentication. */
public class AnonymousAuthenticationProvider implements AuthenticationProvider {
    public CompletableFuture<Void> authenticateRequest(final RequestInformation request, final Map<String, Object> additionalAuthenticationContext) {
        return CompletableFuture.completedFuture(null);
    }    
}

File: abstractions/python/kiota/abstractions/authentication/anonymous_authentication_provider.py
from ..request_information import RequestInformation
from .authentication_provider import AuthenticationProvider


class AnonymousAuthenticationProvider(AuthenticationProvider):
    """This authentication provider does not perform any authentication

    Args:
        AuthenticationProvider (ABC): The abstract base class that this class implements
    """

    async def authenticate_request(self, request: RequestInformation) -> None:
        """Authenticates the provided request information

        Args:
            request (RequestInformation): Request information object
        """
        return
